client/podman/informer.go

```diff
@@ -0,0 +1,148 @@
+package podman
+
+import (
+	"log"
+	"time"
+
+	"github.com/skupperproject/skupper/client/container"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	resyncPeriod = time.Second * 5
+)
+
+//
+// Base types
+//
+
+type informerBase struct {
+	cli          *PodmanRestClient
+	resyncPeriod time.Duration
+}
+
+func (i *informerBase) SetResyncPeriod(t time.Duration) {
+	i.resyncPeriod = t
+}
+
+type updateNotification[T any] struct {
+	oldObj T
+	newObj T
+}
+
+type EventHandler[T any] interface {
+	OnAdd(obj T)
+	OnUpdate(oldObj, newObj T)
+	OnDelete(obj T)
+}
+
+type EventHandlerBase[T any] struct {
+	Add    func(obj T)
+	Update func(oldObj, newObj T)
+	Delete func(obj T)
+}
+
+func (e *EventHandlerBase[T]) OnAdd(obj T) {
+	e.Add(obj)
+}
+
+func (e *EventHandlerBase[T]) OnUpdate(oldObj, newObj T) {
+	e.Update(oldObj, newObj)
+}
+
+func (e *EventHandlerBase[T]) OnDelete(obj T) {
+	e.Delete(obj)
+}
+
+//
+// Container informer
+//
+
+func NewContainerInformer(cli *PodmanRestClient) *ContainerInformer {
```

Comment:
Where does this get used?

Reply:
It will be used as part of the service-controller, which will initially provide event sources to the flow-collector.